### PR TITLE
Add PostgreSQL PCA service

### PR DESCRIPTION
**Pull request by amouhoub**

### CODE REVIEW

File: DataConnectors/resources/catalog/PostgreSQL.xml [hunk 1]
```diff
@@ -311,8 +303,8 @@ print("INSERTING DATA IN POSTGRESQL...")
 print('POSTGRES_HOSTNAME='+HOSTNAME)
 print('POSTGRES_PORT=', PORT)
 print('POSTGRES_DATABASE='+DATABASE)
```
Comment (Aminelouati) [inline DataConnectors/resources/catalog/PostgreSQL.xml]: may be add print('POSTGRES_USER='+USER) as the other variables ?

File: DatabaseServices/resources/catalog/PostgreSQL_Database_Interaction.xml [hunk 2]
```diff
@@ -0,0 +1,509 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<job
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xmlns="urn:proactive:jobdescriptor:3.10"
+     xsi:schemaLocation="urn:proactive:jobdescriptor:3.10 http://www.activeeon.com/public_content/schemas/proactive/jobdescriptor/3.10/schedulerjob.xsd"
+    name="PostgreSQL_Database_Interaction" projectName="PostgreSQL Workflows"
+    priority="normal"
+    onTaskError="continueJobExecution"
+     maxNumberOfExecution="2">
```
Comment (Aminelouati) [inline DatabaseServices/resources/catalog/PostgreSQL_Database_Interaction.xml]: Please remove this ^M everywhere

File: DatabaseServices/resources/catalog/PostgreSQL_Database_Interaction.xml [hunk 3]
```diff
@@ -0,0 +1,509 @@
...
+LABEL = variables.get("LABEL")
+if LABEL:
+    IS_LABELED_DATA='True'
+
+# Please refer to SQLAlchemy doc for more info about database urls.
+# http://docs.sqlalchemy.org/en/latest/core/engines.html#database-urls
+# Never print this to avoid displaying your credentials in the logs
+print("BEGIN Import_Data from " + RDBMS_NAME + " database using " + variables.get("RDBMS_DRIVER") + " connector")
+print("EXECUTING QUERY...")
+print('POSTGRES_HOSTNAME='+HOSTNAME)
```
Comment (Aminelouati) [inline DatabaseServices/resources/catalog/PostgreSQL_Database_Interaction.xml]: print('POSTGRES_USER=', USER)

File: DatabaseServices/resources/catalog/PostgreSQL_Database_Interaction.xml [hunk 4]
```diff
@@ -0,0 +1,509 @@
...
+          <code language="cpython">
+            <![CDATA[
+import pandas as pd
+import numpy as np
+from sqlalchemy import create_engine
+import sys
+
+RDBMS_NAME = 'postgresql'
+RDBMS_DRIVER = variables.get("RDBMS_DRIVER")
+OUTPUT_FILE = None
```
Comment (Aminelouati) [inline DatabaseServices/resources/catalog/PostgreSQL_Database_Interaction.xml]: Please remove this line. The initialization is done after.

File: DatabaseServices/resources/catalog/PostgreSQL_Database_Interaction.xml [hunk 5]
```diff
@@ -0,0 +1,509 @@
...
+    INSERT_MODE = 'append'
+
+# Please refer to SQLAlchemy doc for more info about database urls.
+# http://docs.sqlalchemy.org/en/latest/core/engines.html#database-urls
+# Never print this to avoid displaying your credentials in the logs
+
+print("BEGIN Export_Data to " + RDBMS_NAME + " database using " + variables.get("RDBMS_DRIVER") + " connector")
+print("INSERTING DATA IN POSTGRESQL...")
+print('POSTGRES_HOSTNAME='+HOSTNAME)
+print('POSTGRES_PORT=', PORT)
```
Comment (Aminelouati) [inline DatabaseServices/resources/catalog/PostgreSQL_Database_Interaction.xml]: print('POSTGRES_USER=', USER)

File: PostgreSQL/resources/catalog/PostgreSQL.xml [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+    <variable name="POSTGRES_PASSWORD" value="" />
+    <variable name="POSTGRES_USER" value="postgres" />
+  </variables>
+  <description>
+    <![CDATA[ Deployment of the PostgreSQL Database server. 
+This workflow requires the following third-party credential: {key: posgres://<postgres_username>@<postgres_instance_name>, value: POSTGRES_PASSWORD}. Please refer to the User documentation to learn how to add third-party credentials. ]]>
+  </description>
+  <genericInformation>
+    <info name="bucketName" value="cloud-automation"/>
+    <info name="Documentation" value="https://www.mysql.com/"/>
```
Comment (Aminelouati) [inline PostgreSQL/resources/catalog/PostgreSQL.xml]: doc link: https://www.postgresql.org/

File: DatabaseServices/resources/catalog/Start_PostgreSQL_Service.xml [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+<job
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xmlns="urn:proactive:jobdescriptor:3.10"
+     xsi:schemaLocation="urn:proactive:jobdescriptor:3.10 http://www.activeeon.com/public_content/schemas/proactive/jobdescriptor/3.10/schedulerjob.xsd"
+    name="Start_PostgreSQL_Service" projectName="PostgreSQL"
+    priority="normal"
+    onTaskError="continueJobExecution"
+     maxNumberOfExecution="2">
+  <variables>
+    <variable name="MYSQL_DATABASE" value="" />
```
Comment (Aminelouati) [inline DatabaseServices/resources/catalog/Start_PostgreSQL_Service.xml]: Please check name variables MYSQL -->POSTGRES

File: DataConnectors/resources/catalog/PostgreSQL.xml [hunk 8]
```diff
@@ -84,46 +85,43 @@ from sqlalchemy import create_engine
...
+if not DATABASE:
     print("POSTGRES_DATABASE not defined by the user.")
     sys.exit(1)
-if credentials.get("POSTGRES_USERNAME") is not None and credentials.get("POSTGRES_PASSWORD") is not None:
-    POSTGRES_USERNAME = credentials.get("POSTGRES_USERNAME")
-    POSTGRES_PASSWORD=credentials.get("POSTGRES_PASSWORD")
-else:
-    print("You first need to add third-party credentials (POSTGRES_USERNAME and POSTGRES_PASSWORD) for the database in the scheduler-portal.")
+if not USER:
+    print("POSTGRES_USER not defined by the user.")
```
Comment (Aminelouati) [inline DataConnectors/resources/catalog/PostgreSQL.xml]: I would like this error message: ERROR: The POSTGRES_USER variable is not provided by the user.

File: DatabaseServices/resources/catalog/PostgreSQL_Database_Interaction.xml (hunk 2 above)
Comment (amouhoub) [reply DatabaseServices/resources/catalog/PostgreSQL_Database_Interaction.xml]: It is only here 😄 done

File: DataConnectors/resources/catalog/PostgreSQL.xml (hunk 1 above)
Comment (amouhoub) [reply DataConnectors/resources/catalog/PostgreSQL.xml]: that's a good point thank you

File: DatabaseServices/resources/catalog/PostgreSQL_Database_Interaction.xml (hunk 5 above)
Comment (amouhoub) [reply DatabaseServices/resources/catalog/PostgreSQL_Database_Interaction.xml]: done but does not appear here

**Pull request merged**